Draft docs page for processing levels and updated processing_level_url

emiliom opened the pull request:
**Draft** Data Processing Levels page. Incorporates the content from https://github.com/uw-echospace/data-processing-levels/blob/main/data_processing_levels.md and adds an introductory paragraph.

In the Table of Content, I've placed this page at the end of the User Guide section.

Set the `processing_level_url` global attribute to https://echopype.readthedocs.io/en/stable/processing-levels.html

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,61 @@
+
+The decades-long experience from the satellite remote sensing community has shown that a set of robust and well-articulated definitions of "data processing levels" [1]–[5] can lead directly to broad and highly productive use of the data. Processing level designations also provide important context for data interpretation [6]. However, no such community agreement exists for active acoustic data, a ...
+
+The `echopype` team is developing a clearly defined progression of data processing levels for active ocean sonar data, leveraging the accumulated experience from remote sensing and large-scale, long-term ocean and ecological observatories [7]–[10]. Data processing functions in `echopype` are clearly associated with specific Processing Level inputs and outputs, and when appropriate, will generate  ...
```
```suggestion
The `echopype` team is developing a clearly defined progression of data processing levels for active ocean sonar data. The development leverages the collective experience from remote sensing and large-scale, long-term ocean and ecological observatories [7]–[10]. Data processing functions in `echopype` are clearly associated with specific Processing Level inputs and outputs, and when appropriate, will generate a `processing_level` dataset global attribute with entries such as "Level 1A", "Level 2B", "Level 4", etc.
```

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,61 @@
+
+### Level 1 (L1)
+
+**Description:** Raw data packaged with ancillary information and converted and standardized to an open convention and standard data formats. May be distributed as sets of individual converted files as originally segmented into arbitrary time ranges during sensor file creation or compiled into larger granules corresponding to logical deployment intervals.
```
```suggestion
**Description:** Raw data packaged with ancillary information and converted and standardized to an open convention and standard data formats. May be distributed in the following two forms: 
  - as sets of individual converted files as originally segmented into arbitrary time ranges during sensor file creation, or
  - compiled into larger granules corresponding to logical deployment intervals.
```

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,61 @@
+
+**Description:** Raw data packaged with ancillary information and converted and standardized to an open convention and standard data formats. May be distributed as sets of individual converted files as originally segmented into arbitrary time ranges during sensor file creation or compiled into larger granules corresponding to logical deployment intervals.
+
+- **L1A**: Raw L0 data converted to a standardized, open format with geographic coordinates (latitude & longitude) processed and included. Includes other ancillary information extracted from sensor-generated L0 data or other external sources. May include environmental information such as temperature, salinity and pressure. Use of the SONAR-netcDF4 v1 convention is strongly recommended.
```
In "Raw L0 data converted to a standardized, open format with geographic coordinates (latitude & longitude) processed and included." it is not clear what "processed" means.

emiliom replied:
I've removed "processed"

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,61 @@
+
+- `Sv` resampled to a common, high-resolution grid across channels
+- Mean Volume Backscattering Strength (`MVBS`)
+- `Sv` frequency difference across two channels resampled to a common grid
```
I am not sure why the frequency difference between Sv from two channels is a data product. It is more of a processing step to classify?  Do you mean the masked Sv afterwards?

emiliom replied:
I agree. I've made a suggestion to emphasize the mask application, and retaining frequency differencing only as a mask example. How does that look?

leewujung replied:
I think that would work for this iteration. Let's discuss where masks should belong as data products -- i feel it may belong to L4, but we can reference other examples to determine.

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,61 @@
+
+**Description:** Acoustically derived biological features, involving further processing of L3 data that may include data reduction or incorporation of external sources of data.
+
+- Nautical Area Backscattering Coefficients (`NASC`)
```
```suggestion
- Nautical Area Backscattering Coefficients (`NASC`), potentially partitioned into inferred sources of scattering
```

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,61 @@
+- Nautical Area Backscattering Coefficients (`NASC`)
+- Summary statistics of echogram features (center_of_mass, dispersion, etc)
+- Taxon or species-level data labels (classification). May originate from a variety of methods, including frequency difference thresholds.
+- Biomass estimation
```
```suggestion
- Estimated biomass, combined or partitioned into different taxonomic sources.
```

leewujung reviewed (commented):
I added some comments/suggestion edits.

This is nitpicking but we should standardize whether or not each sentence ends with a `.`?

emiliom commented:
Thanks. I'll follow up later.

emiliom commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,63 @@
+  - as sets of individual converted files as originally segmented into arbitrary time ranges during sensor file creation, or
+  - compiled into larger granules corresponding to logical deployment intervals.
+
+- **L1A**: Raw L0 data converted to a standardized, open format with geographic coordinates (latitude & longitude) processed and included. Includes other ancillary information extracted from sensor-generated L0 data or other external sources. May include environmental information such as temperature, salinity and pressure. Use of the SONAR-netcDF4 v1 convention is strongly recommended.
```
I'm removing "processed".
```suggestion
- **L1A**: Raw L0 data converted to a standardized, open format with geographic coordinates (latitude & longitude) included. Includes other ancillary information extracted from sensor-generated L0 data or other external sources. May include environmental information such as temperature, salinity and pressure. Use of the SONAR-netcDF4 v1 convention is strongly recommended.
```

emiliom commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,63 @@
+
+- `Sv` resampled to a common, high-resolution grid across channels
+- Mean Volume Backscattering Strength (`MVBS`)
+- `Sv` frequency difference across two channels resampled to a common grid
```
```suggestion
- Masks applied to `Sv` based on different potential criteria, sources or calculations, such as frequency difference across two channels.
```

emiliom commented:
I've accepted or addressed all your comments. Thanks! I left one question open for you to look at, with a suggestion. Please decide on my suggestion, or make a new one, and move forward with this PR.

As for this:

> This is nitpicking but we should standardize whether or not each sentence ends with a `.`?

Let's deal with this in a second pass, so you can go ahead with creating the release.

leewujung commented on docs/source/processing-levels.md:
```diff
@@ -0,0 +1,63 @@
+
+- `Sv` resampled to a common, high-resolution grid across channels
+- Mean Volume Backscattering Strength (`MVBS`)
+- Masks applied to `Sv` based on different potential criteria, sources or calculations, such as frequency difference across two channels.
```
```suggestion
- Masks applied to `Sv` based on different scattering source classification criteria and methods, such as frequency difference across two channels.
```
